web/src/engine/websites/MeianPlus.ts

```diff
@@ -0,0 +1,160 @@
+import { Tags } from '../Tags';
+import icon from './MeianPlus.webp';
+import { Chapter, DecoratableMangaScraper, type MangaPlugin, Manga, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Fetch, FetchWindowScript } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+import { Exception } from '../Error';
+import { WebsiteResourceKey as R } from '../../i18n/ILocale';
+
+type APISeries = {
+    licences: APISerieDetails[];
+}
+
+type APISerie = {
+    licence: APISerieDetails
+}
+
+type APISerieDetails = {
+    id_licence: number,
+    titre_licence: string,
+    articles: {
+        ref: number,
+        titre: string,
+        ebook_statut: boolean
+    }[]
+}
+
+type APIEbook = {
+    success: boolean,
+    message: string,
+    ebook: {
+        max_page: number
+    }
+}
+
+type APIImages = {
+    images: {
+        ref: number,
+        key: string,
+        w: number,
+        h: number,
+        param: string
+    }[]
+}
+
+type BlockInfo = {
+    w: number,
+    h: number
+}
+
+const tokenScript = `decodeURIComponent(document.cookie.match(/token_meian_plus=([^;]+)/).at(1)).replaceAll('"', '');`;
```

Comment:
Maybe this can be simplified
`cookieStore.get('token_meian_plus')).then(cookie => cookie.value);`

Reply:
That website is somehow proxying cookiestore and its ugly to work with ; ofc i would have used cookiestore instead.

Reply:
So filtering all cookies does also not work?
`cookieStore.getAll().then(cookies => cookies.find(cookie => cookie.name === 'token_meian_plus')?.value);`

Reply:
The `cookieStore` works for me, when evaluating the promise.